metron-streaming/Metron-Solr/src/main/java/org/apache/metron/writer/solr/SolrWriter.java

```diff
@@ -65,7 +71,9 @@ public void write(String sourceType, Configurations configurations, List<Tuple>
       }
       UpdateResponse response = solr.add(document);
     }
-    solr.commit(getCollection(configurations));
+    if (shouldCommit) {
+      solr.commit(getCollection(configurations));
+    }
   }
```

Comment:
There may be a little more work to do on this. I ran it up last night and after 1/2 an hour had to issue a manual commit to see any documents in Solr. IIRC, openSearcher is false by default which would explain that.